app/models/miq_region_remote.rb

```diff
@@ -84,7 +84,7 @@ def self.with_remote_connection(host, port, username, password, database, adapte
     host = host.to_s.strip
     raise ArgumentError, _("host cannot be blank") if host.blank?
     if [nil, "", "localhost", "localhost.localdomain", "127.0.0.1", "0.0.0.0"].include?(host)
-      local_database = ActiveRecord::Base.configurations.fetch_path(Rails.env, "database").to_s.strip
+      local_database = (ActiveRecord::Base.configurations[Rails.env] || {})["database"].to_s.strip
```

Comment:
I assume configurations is no longer a Hash/HWIA?  If not, would it make sense to mixin in MoreCoreExtensions::Nested into the new class (or is it better to just _not_ treat it like a hash anymore as is here?)

Reply:
Is there a fetch method on configurations allowing something like...

```suggestion
      local_database = ActiveRecord::Base.configurations.fetch(Rails.env, {})["database"].to_s.strip
```

Reply:
I think my commit message covers this pretty well (not linking since this PR will probably be rebased again and will become a broken link):

```
 [rails6][EvmDatabase] Fix .host

Rails 6 changed database configuration to support multi-db environments:

rails/rails#33637

So how one accesses the `@config` hash has changed.

The original error suggested the following:

    NotImplementedError: `ActiveRecord::Base.configurations` in Rails 6
    now returns an object instead of a hash. The `fetch_path` method is
    not supported. Please use `configs_for` or consult the documentation
    for supported methods.

However, there is no `[]` or `host` method, so it would require calling
`configs_for(env_name: ENV["RAILS_ENV"]).config['host']`.

This solution seemed like the more terse solution.
```

Reply:
Wait... I misread... let me see if `.fetch` will work for `.configurations`

Reply:
Gah sorry...since i review in aggregate, I never even think to look at commit messages for specific comments (I'm not even sure how to get to a specific commits from the /files github overview page).

Reply:
I wonder if upstream rails would accept a patch for a .fetch method

Reply:
> (I'm not even sure how to get to a specific commits from the /files github overview page).

In a PR, you can go to a specific commit, the `[Gemfile] Rails 6.0` in this PR:

https://github.com/ManageIQ/manageiq/pull/20778/commits/9bc9d780d2e254b2a045f13b76e7c8505fda37ca

Which will give you a link like the above in the format of:

```
/pull/[PR#]/commits/[COMMIT_SHA]
```

Which has a `< Prev` and `Next >` buttons at the top to allow you to go through each commit one by one for a given PR.

Reply:
> I wonder if upstream rails would accept a patch for a .fetch method

For this one, I am doubtful since they explicitly said we aren't a hash any more and even have things like `.to_h` deprecated on `master`:

https://github.com/rails/rails/blob/master/activerecord/lib/active_record/database_configurations.rb#L103-L109

So I am going to go out on a limb and say 'no' on that one.

Reply:
`ENV[:key] || {}` is pretty idiomatic. Is there a reason to go to words like `ENV.fetch(:key, {})`? I can't read the second one as well as the first one.

I also don't know that we want to take advantage of the `fetch` nuance - I think we want to have a `{}` whether it is defined as nil or not defined.

Reply:
> . Is there a reason to go to words like ENV.fetch(:key, {})

Yes, because it doesn't introduce extra parenthesis in chaining...compare

```ruby
ENV.fetch(:key, {}).first
vs
(ENV[:key] || {}).first
```

Even so, it's a moot point because the configurations object doensn't support .fetch anyway.

Reply:
Going to resolve this one, since I think this is the best option unless we really want to go with:

```ruby
ActiveRecord::Base.configurations.configs_for(env_name: ENV["RAILS_ENV"]).config['database'].to_s.strip
```

v.s.

```ruby
(ActiveRecord::Base.configurations[Rails.env] || {})["database"].to_s.strip
```